### PR TITLE
Scottx611x/remove workflow files dl model

### LINKED ISSUE #2173: Remove the `WorkflowFilesDL` model
First off, `WorkflowFilesDL` is not a great name. If we do need it, lets update that.

This model keeps track of Galaxy Dataset's filenames and which Workflow step they correspond to which is then used later on to aid in downloading said datasets, but seems to only need to stick around for legacy `Analysis` runs.

I'm not convinced we need this model at all for new `WorkflowTool`-based `Analysis` runs.

- [x] Look into this here: https://github.com/refinery-platform/refinery-platform/blob/develop/refinery/analysis_manager/tasks.py#L640-L724

---

- [x] Remove code creating `WorkflowFilesDL` instances
- [x] Remove afformentioned code
- [x] Remove ManyToMany in `Analysis`
- [x] Remove `WorkflowFilesDL` Model & Admin interface
- [x] Remove tests

### PR DESCRIPTION
**Pull request by scottx611x**
Closes #2173 

- [x]  Test this on an instance with beta.scc snapshots and check how existing Analyses are affected

### CODE REVIEW

**Comment by scottx611x** [comment]
Failing build  above from cypress test: `Satori Visualization Sorts nodes in list graph by name`
See: #2348

**Comment by scottx611x** [comment]
@hackdna not sure. I'll test this on an instance with `beta.scc` snapshots and check

**Comment by scottx611x** [comment]
@hackdna I've gotten around to testing this out w/ beta data and existing Analyses and their data aren't affected.

The idea behind `WorkflowFilesDL` was to be a temporary placeholder for derived data file info until an actual `FileStoreitem` got created. After the FSI creation it was a non-essential entity

File: refinery/analysis_manager/tasks.py [hunk 1]
```diff
@@ -464,19 +464,8 @@ def _get_galaxy_download_task_ids(analysis):
 
     # retrieving list of files to download for workflow
     tool = _get_workflow_tool(analysis.uuid)
-    tool.create_workflow_file_downloads()
     tool.create_analysis_output_node_connections()
 
-    dl_files = analysis.workflow_dl_files
-    # creating dictionary based on files to download predetermined by workflow
-    # w/ keep operators
-    dl_dict = {}
```
Comment (mccalluc) [inline refinery/analysis_manager/tasks.py]: huh: This dict didn't seem to be used, even before your changes? Just vestigial, probably?

Comment (scottx611x) [reply refinery/analysis_manager/tasks.py]: @mccalluc It was used way back when we were launching analyses the old way, but looking at this code in particular was the motivator for the removal of the whole model

**Pull request merged**